components/button/index.js

```diff
@@ -25,28 +29,60 @@ export function Button( props, ref ) {
 		className,
 		disabled,
 		focus,
+		children,
 		...additionalProps
 	} = props;
 
+	const tag = href !== undefined && ! disabled ? 'a' : 'button';
+	const tagProps = tag === 'a' ? { href, target, rel } : { type: 'button', disabled };
+	const { icon = true } = additionalProps;
```

Comment:
As I see it, `additionalProps` are those which are not explicitly handled by the component itself. In this case, we _are_ handling `icon`, and it should be included in the destructuring which occurs at the top of this function.

Reply:
We should document `icon`. In particular, it's unclear to me whether this is always a boolean (where I might expect a prefix to indicate this such as `hasIcon`, `showIcon`) or an overloaded value where a boolean overrides a default.